Fix cache_free fucntion

qwe661234 opened the pull request:
After we introduced the memory pool for blocks and IRs, we don't need to free the memory of blocks and IRs by cache.

jserv reviewed (changes requested):
Fix undefined behavior reported by the optimizer:
```
src/cache.c:521:26: warning: iteration 4 invokes undefined behavior [-Waggressive-loop-optimizations]
  521 |         free(cache->lists[i]);
      |              ~~~~~~~~~~~~^~~
src/cache.c:520:23: note: within this loop
  520 |     for (int i = 0; i < THRESHOLD; i++)
      |                       ^
```

qwe661234 committed:
Fix cache_free fucntion

After we introduced the memory pool for blocks and IRs, we don't need
to free the memory of blocks and IRs by cache.
